src/test/java/com/example/joy/myFirstSpringBoot/controllers/BirthdayInfoControllerIT.java

```diff
@@ -0,0 +1,112 @@
+package com.example.joy.myFirstSpringBoot.controllers;
+
+import com.example.joy.myFirstSpringBoot.services.BasicBirthdayService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.user;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ContextConfiguration(classes = { BirthdayInfoController.class,
```

Comment:
ugg tabs...
I ran into problems between the context setup by the `ContextConfiguration` and `WebMvcTest` annotations.
Without setting it up like this i was getting 404's because the controller was not in the context (I have no idea why this was the case)